test/integration/endpoints/endpoints_test.go

```diff
@@ -20,27 +21,65 @@ func TestMain(m *testing.M) {
 }
 
 func TestGoodEndpoints(t *testing.T) {
-	ns := TestHelper.GetLinkerdNamespace()
-	vizNs := TestHelper.GetVizNamespace()
+	ctx := context.Background()
+	controlNs := TestHelper.GetLinkerdNamespace()
 	testDataPath := "testdata"
 	cmd := []string{
 		"diagnostics",
 		"endpoints",
-		fmt.Sprintf("linkerd-dst.%s.svc.cluster.local:8086", ns),
-		fmt.Sprintf("grafana.%s.svc.cluster.local:3000", vizNs),
-		fmt.Sprintf("linkerd-identity.%s.svc.cluster.local:8080", ns),
-		fmt.Sprintf("linkerd-proxy-injector.%s.svc.cluster.local:443", ns),
-		fmt.Sprintf("tap.%s.svc.cluster.local:8088", vizNs),
-		fmt.Sprintf("web.%s.svc.cluster.local:8084", vizNs),
+		fmt.Sprintf("linkerd-dst.%s.svc.cluster.local:8086", controlNs),
+		fmt.Sprintf("linkerd-identity.%s.svc.cluster.local:8080", controlNs),
+		fmt.Sprintf("linkerd-proxy-injector.%s.svc.cluster.local:443", controlNs),
 	}
 
 	if !TestHelper.ExternalPrometheus() {
-		cmd = append(cmd, fmt.Sprintf("prometheus.%s.svc.cluster.local:9090", vizNs))
+		cmd = append(cmd, fmt.Sprintf("nginx.%s.svc.cluster.local:8080", "linkerd-endpoints-test"))
+		TestHelper.WithDataPlaneNamespace(ctx, "endpoints-test", map[string]string{}, t, func(t *testing.T, ns string) {
+			nginx, err := TestHelper.LinkerdRun("inject", "testdata/nginx.yaml")
+			if err != nil {
+				testutil.AnnotatedFatal(t, "unexpected error", err)
+			}
+
+			out, err := TestHelper.KubectlApply(nginx, ns)
+			if err != nil {
+				testutil.AnnotatedFatalf(t, "unexpected error", "unexpected error: %v output:\n%s", err, out)
+			}
+
+			err = TestHelper.CheckPods(ctx, ns, "nginx", 1)
+			if err != nil {
+				if rce, ok := err.(*testutil.RestartCountError); ok {
+					testutil.AnnotatedWarn(t, "CheckPods timed-out", rce)
+				} else {
+					testutil.AnnotatedError(t, "CheckPods timed-out", err)
+				}
+			}
+
+			checkEndpoints(t, cmd, testDataPath, controlNs, ns)
+		})
 	} else {
 		cmd = append(cmd, "prometheus.external-prometheus.svc.cluster.local:9090")
 		testDataPath += "/external_prometheus"
+		checkEndpoints(t, cmd, testDataPath, controlNs, "external-prometheus")
```

Comment:
Isn't the external-prometheus case only relevant for the viz's suite, so not to be considered here?

Reply:
Yup, you're right. I was thinking about this too. I think under our current plan to create better isolated test suites, we'd have to duplicate this test to have it present in both, `external` and `deep`. 

I don't think that's a really good idea, if we check the endpoints in `deep` then we can be pretty sure it's going to be the same regardless of what deployment in what namespace we have. From my perspective, having this endpoints test for external resources as well is a bit of a waste, we don't test anything new. I deleted the external prometheus part, I'm fine with testing just the control plane and nginx.

Wdyt?

Reply:
Agreed, no benefit from having it in external resources too :+1: